src/libraries/Common/src/System/Net/CookieParser.cs

```diff
@@ -791,7 +792,7 @@ internal Cookie GetServer()
                                     if (!domainSet)
                                     {
                                         domainSet = true;
-                                        cookie.Domain = CheckQuoted(_tokenizer.Value);
+                                        cookie!.Domain = CheckQuoted(_tokenizer.Value);
```

Comment:
Same question; what prevents cookie from being null?  If we can provably say it's not null, seems better to assert that in one place and then we can avoid all of these !s.